app/src/main/java/org/dicekeys/app/openpgp/SecretPacket.kt

```diff
@@ -0,0 +1,57 @@
+package org.dicekeys.app.openpgp;
+
+import com.google.common.io.ByteStreams
+import org.bouncycastle.crypto.params.Ed25519PrivateKeyParameters
+import java.math.BigInteger
+import java.nio.ByteBuffer
+import java.security.MessageDigest
+
+class SecretPacket(private val privateKey: ByteArray, private val timestamp: UInt) : Packet() {
+
+    private val privateKeyEd255119 = Ed25519PrivateKeyParameters(privateKey, 0)
+
+    override val pTag: Int
+        get() = 0x94
+
+    val publicPacket: PublicPacket by lazy {
+        PublicPacket(privateKeyEd255119.generatePublicKey().encoded, timestamp)
+    }
+
+    override val body: ByteArray by lazy {
+        val body = ByteStreams.newDataOutput()
+
+        val taggedPublicKey = byteArrayOf(0x40) + privateKeyEd255119.generatePublicKey().encoded
```

Comment:
Is there a pointer to an RFC or spec for the private key format?

Where does one go to see the specs this binary format is based on?

Reply:
It's an uncompressed point. I think DER encoding has something similar.

> 13.3.  EdDSA Point Format

> The EdDSA algorithm defines a specific point compression format.  To
   indicate the use of this compression format and to make sure that the
   key can be represented in the Multiprecision Integer (MPI) format the
   octet string specifying the point is prefixed with the octet 0x40.
   This encoding is an extension of the encoding given in [SEC1] which
   uses 0x04 to indicate an uncompressed point.

> For example, the length of a public key for the curve Ed25519 is 263
   bit: 7 bit to represent the 0x40 prefix octet and 32 octets for the
   native value of the public key.